### ISSUE TITLE
Internal: Add tests

### ISSUE THREAD

**Issue by kaicataldo** [kaicataldo/eleventy-plugin-react#10]
Now that the feature set is stabilizing a bit, the plugin desperately needs tests. Any help is appreciated!

**Comment by dpfavand** [comment]
Do you have any preferences for testing library? 

Jest + [testing-library/react](https://testing-library.com/docs/react-testing-library/intro) seems like a pretty common combination that probably has a relatively low learning curve for contributors.

**Comment by kaicataldo** [comment]
Yeah, that does seem pretty common in the React world! Makes sense to test the `withHydration` HOC and the hydrated bundle. What I was initially thinking was:

1. Unit testing `withHydration`
2. Integration tests where we run the Eleventy build (maybe a nested full project with a package.json and everything?) and then assert that the generated files have the expected content. We could then load up the site in a headless browser and make assertions about the hydrated component behavior.

Thoughts?
